### PR TITLE
remove subaccounts with no asset/perpetual positions

### PR DESCRIPTION
**Pull request by jayy04**

### CODE REVIEW

**Comment by teddyding** [comment]
Coul you fix failing subaccounts e2e test?

File: protocol/x/subaccounts/keeper/grpc_query_subaccount_test.go [hunk 1]
```diff
@@ -93,7 +94,7 @@ func TestSubaccountQuerySingle(t *testing.T) {
 func TestSubaccountQueryPaginated(t *testing.T) {
 	ctx, keeper, _, _, _, _, _, _ := keepertest.SubaccountsKeepers(t, true)
 	wctx := sdk.WrapSDKContext(ctx)
-	msgs := createNSubaccount(keeper, ctx, 5)
+	msgs := createNSubaccount(keeper, ctx, 5, big.NewInt(1_000))
```
Comment (teddyding) [inline protocol/x/subaccounts/keeper/grpc_query_subaccount_test.go:97]: Could you add a unit test that sets a subaccount with empty positions?

File: protocol/x/subaccounts/keeper/subaccount.go [hunk 2]
```diff
@@ -25,10 +25,16 @@ import (
 // SetSubaccount set a specific subaccount in the store from its index.
```
Comment (lucas-dydx) [inline protocol/x/subaccounts/keeper/subaccount.go:25]: Suggestion: update comment to note empty subaccounts are removed

**Review by teddyding — APPROVED** [review]
